### DISCUSSION TITLE
Flatpak and GTK4 apps

### DISCUSSION THREAD

**Discussion by Flashwalker** [yshui/picom#1014] [Q&A]
Any solution to fix **Flatpak** and **GTK4** apps?

Flatpak:
![2023-01-29_20-28-01](https://user-images.githubusercontent.com/2188215/215330059-2cb9e4d1-08a0-42f0-a5e6-7cc46110c40d.jpg)
![2023-01-29_20-25-38](https://user-images.githubusercontent.com/2188215/215330068-28643f59-a5a1-42c8-9205-0557d4997b52.jpg)

They seem to have their own shadows... or something

For **GTK4** file dialog i set exclusion:
```
,"class_g = 'xdg-desktop-portal-gnome'" # GTK4 file dialog. -shadow, -blur
```

for now Gnome gayvelopers uses GTK4 for xdg-desktop-portal file dialog in Flatpak apps:  

![2023-01-29_20-35-45](https://user-images.githubusercontent.com/2188215/215330020-40005bb3-7bdd-46ce-bf65-aacfc91ff0ed.jpg)
![2023-01-29_20-35-28](https://user-images.githubusercontent.com/2188215/215330015-f48f4d39-7a06-443a-a53f-aa54365d1bd5.jpg)

**Comment by Flashwalker** [comment]
The only way i found for now is to whitelist certain apps and exclude the rest:
E.g.:
```
# Blur: whitelist:
blur-background-exclude = [
  "class_g != 'URxvt' &&
   class_g != 'Alacritty' &&
   class_g != 'st-256color' &&
   class_g != 'tabbed' &&
   class_g != 'kitty' &&
   class_g != 'Xfce4-terminal' &&
   class_g != 'Notification' &&
   window_type != 'tooltip' &&
   window_type != 'menu' &&
   window_type != 'popup_menu' &&
   window_type != 'dropdown_menu' &&
   window_type != 'splash' &&
   window_type != 'combo'"
   
   ,"name = 'Whisker Menu'"
   ,"name = 'xfce4-panel-plug'"
];
```

**Comment by Flashwalker** [comment]
or just:
```
wintypes : 
{
  normal = {
    blur-background = false;
  };
};
```

**Comment by absolutelynothelix** [comment]
afaik you need to exclude `_NET_WM_FRAME_EXTENTS` or/and `_NET_FRAME_EXTENTS` atoms. i haven't done that myself so i can't give you an example of this, but you can read the manual or search through the issues for a possible example.
